File: src/modules/algorithm/seed/algorithms/graph-list-dfs/javascript/solution-recursive.js
/**
 * Performs a Depth-First Search (DFS) traversal on a graph using recursion.
 *
 * @param {Object} graph - An adjacency list representation of the graph where keys are vertices and values are arrays of adjacent vertices
 * @param {string} startVertex - The vertex to start the DFS traversal from
 * @returns {string[]} An array containing the vertices in the order they were visited
 */
function dfs(graph, startVertex) {
  const visited = new Set();
  const result = [];

  // Helper function for recursive DFS
  function dfsHelper(vertex) {
    // Mark the current vertex as visited and add to result
    visited.add(vertex);
    result.push(vertex);

    // Visit all adjacent vertices that haven't been visited yet
    for (const neighbor of graph[vertex]) {
      if (!visited.has(neighbor)) {
        dfsHelper(neighbor);
      }
    }
  }

  // Start DFS from the given vertex
  dfsHelper(startVertex);

  return result;
}

module.exports = { dfs };

File: src/modules/algorithm/seed/algorithms/graph-list-dfs/python/solution_recursive.py
def dfs(graph, start_vertex):
    """
    Performs a Depth-First Search (DFS) traversal on a graph using recursion.
    
    Args:
        graph: A dictionary representing the graph as an adjacency list
               where keys are vertices and values are lists of adjacent vertices
        start_vertex: The vertex to start the DFS traversal from
        
    Returns:
        A list containing the vertices in the order they were visited
    """
    visited = set()
    result = []
    
    def dfs_helper(vertex):
        # Mark the current vertex as visited and add to result
        visited.add(vertex)
        result.append(vertex)
        
        # Visit all adjacent vertices that haven't been visited yet
        for neighbor in graph[vertex]:
            if neighbor not in visited:
                dfs_helper(neighbor)
    
    # Start DFS from the given vertex
    dfs_helper(start_vertex)
    
    return result 